buffer/src/main/java/io/netty/buffer/CompositeByteBuf.java

```diff
@@ -413,39 +408,34 @@ private int addComponents0(boolean increaseWriterIndex, final int cIndex, ByteBu
      * {@link CompositeByteBuf}.
      */
     public CompositeByteBuf addComponents(int cIndex, Iterable<ByteBuf> buffers) {
-        addComponents0(false, cIndex, buffers);
-        consolidateIfNeeded();
-        return this;
+        return addComponents(false, cIndex, buffers);
     }
 
     // TODO optimize further, similar to ByteBuf[] version
     // (difference here is that we don't know *always* know precise size increase in advance,
     // but we do in the most common case that the Iterable is a Collection)
-    private int addComponents0(boolean increaseIndex, int cIndex, Iterable<ByteBuf> buffers) {
+    private CompositeByteBuf addComponents(boolean increaseIndex, int cIndex, Iterable<ByteBuf> buffers) {
         if (buffers instanceof ByteBuf) {
             // If buffers also implements ByteBuf (e.g. CompositeByteBuf), it has to go to addComponent(ByteBuf).
-            return addComponent0(increaseIndex, cIndex, (ByteBuf) buffers);
+            return addComponent(increaseIndex, cIndex, (ByteBuf) buffers);
         }
         checkNotNull(buffers, "buffers");
         Iterator<ByteBuf> it = buffers.iterator();
         try {
             checkComponentIndex(cIndex);
 
             // No need for consolidation
-            while (it.hasNext()) {
-                ByteBuf b = it.next();
-                if (b == null) {
-                    break;
-                }
+            for (ByteBuf b; it.hasNext() && (b = it.next()) != null;) {
                 cIndex = addComponent0(increaseIndex, cIndex, b) + 1;
```

Comment:
Seems this should be `addComponent()` (not 0).

Reply:
@ejona86 this wasn't changed in this PR, but I think it was intended to be the 0 version as written. The buffer `b` is certain to be non-null, and we're avoiding consolidation after each addition so that it can be done once at the end of the enclosing method.

Reply:
Oh, I missed the `(b = it.next()) != null)`. I thought it was the normal `b = it.next()`. It doesn't seem the loop is any better with this change and I might argue it was more clear before. (It's not clear why it would choose to stop early if `b == null`, but it was obvious it was explicitly handling that case.)